presto-docs/src/main/sphinx/connector/hive-cache.rst

```diff
@@ -0,0 +1,76 @@
+===================
+Hive Object Caching
+===================
+
+Querying object storage with the Hive connector is a very common use case for
+Presto, that often involves the transfer of large amounts of data. The objects
+are retrieved from HDFS, or any other supported object storage, by multiple
+workers and processed on these workers.
+
+Repeated queries with different parameters, or even different queries from
+different users, often access, and therefore transfer, the same objects. The
+Hive object caching can provide significant performance benefits, by avoiding
+the repeated network transfers and instead accessing copies of the objects from
+a local cache.
+
+.. warning::
+  Hive object caching is currently available as beta release only.
+
+.. contents::
+  :local:
+
+Architecture
+------------
+
+The Hive object caching provides a read-through cache. After first retrieval
+from storage by any query, objects are cached in the local cache storage on the
+workers. Objects are cached on local storage of each worker and managed by a
+bookkeeper component. The cache chunks are 1MB in size and are well suited for
```

Comment:
is this caching done by  rubix, alluxio or something else?

Reply:
It is done by Rubix as can be seen in the replaced PR. From my understanding the intention of the implementation is to hide this fact as an implementation detail but I might be wrong. Hence there is an ongoing discussion on where to take this PR on what to add and what to leave out.